File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.js
/**
 * @param {string[]} words
 * @return {string}
 */

var firstPalindrome = function(words) {
    const isPal = (word) =>{
    var l = 0, r = word.length-1;
        while(l <= r ){
            if(word[l] != word[r]) return false;
            l++;
            r--;
        }
        return true;
    }
    for(const word of words){
        if(isPal(word)){
            return word;
        }
    }
    return "";
};

File: 2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        def isPal(w):
            l = 0
            r = len(w)-1
            while l < r :
                if(w[l] != w[r]):
                    return False
                l += 1
                r -= 1
            return True
        
        for w in words:
            if(isPal(w)):
                return w
        return ""